src/test/ui/type-alias-impl-trait/generic_nondefining_use.stderr

```diff
@@ -4,13 +4,17 @@ error: at least one trait must be specified
 LL | type Cmp<T> = impl 'static;
    |               ^^^^^^^^^^^^
 
-error: defining opaque type use does not fully define opaque type: generic parameter `T` is specified as concrete type `u32`
-  --> $DIR/generic_nondefining_use.rs:10:1
+error: non-defining opaque type use in defining scope
```

Comment:
This error message seems pretty confusing to me. But I guess the old one was pretty bad too so I don't think it needs to block this PR...

Reply:
I used the error that used to be only emitted for lifetimes, also for types and consts.

I don't like either error but I didn't want to make a third variant of my own.